src/Skills/Systems/LightningCannonSystem.gd

```diff
@@ -0,0 +1,41 @@
+extends "res://Ecs/Core/System.gd"
+
+var _event_bus
+
+func _init(event_bus):
+	required_component_types = ["HealthComponent"]
+	_event_bus = event_bus
+	_event_bus.connect("lightning_cannon", self, 'activate')
+
+func activate(source_entity, target: Vector2):
+	var target_entity = _get_entity_at(target)
+	if target_entity != null and source_entity != target_entity:
+		_lightning_cannon(source_entity, target_entity, Constants.LIGHTNING_CANNON_DAMAGE, Constants.LIGHTNING_CANNON_RADIUS)
+		_event_bus.emit_signal("end_turn")
+
+func _lightning_cannon(source, target, damage, radius):
+	if damage <= 0 or radius <= 0:
+		return
+	
+	print("cannon: " + str(damage) + ", " + str(radius))
+	_event_bus.emit_signal("damage_entity", target, damage)
+	var target_2 = _get_entity_within_radius(target.position, radius)
+	if target_2 != null and target_2 != target and source != target_2:
```

Comment:
I like deterministic difficult roguelikes. I thought this would work as "hit everything in range." I see you have coded "hit one of something in range" which works too.

Reply:
More like "hit everything in range until 'charge' runs out". 

- Lightning jumps from enemy to enemy.
- Jumping decreases damage dealt and the radius to jump within, according to distance.
- Once either the damage or the radius reach a non-positive value, or there are no more enemies in the remaining radius, that's when the attack ends. 

I think the effect is pretty cool. It can just jump around between two enemies, but unless they're next to each other, the radius nerf will take its toll after the first or second jump.